### PR TITLE
Massive refactor, wrote testing framework, ported to flask-restplus

### PR DESCRIPTION
**Pull request by ZeroEcks**

### CODE REVIEW

File: app/user/api.py [hunk 1]
```diff
@@ -1,17 +1,23 @@
-from .models import User
+import re
+
+from flask import current_app
 from flask_restful import Resource, reqparse
 from sqlalchemy.exc import IntegrityError
-from flask import current_app
-import re
+
+from .models import User
```
Comment (me4502) [inline app/user/api.py]: Do we have a proper styleguide for imports?

**Comment by KnightHawk3** [comment]
Yeah, run isort over it and use what it says.

File: app/user/api.py (hunk 1 above)
Comment (ZeroEcks) [reply app/user/api.py]: Yeah just run isort over it and use what it says ;)

**Comment by ZeroEcks** [comment]
Thanks guys, i'll be here all night.

**Pull request merged**